pkg/cache/cache.go

```diff
@@ -44,6 +44,17 @@ var (
 	defaultSyncPeriod = 10 * time.Hour
 )
 
+// InformerGetOption defines an option that alters the behavior of how informers are retrieved.
+type InformerGetOption func(*internal.GetOptions)
```

Comment:
Is it really okay to have an internal type as part of an exported function?

Reply:
I didn't find anything directly on-point, but exporting public constants that use private classes is a best practice for context keys: https://forum.golangbridge.org/t/the-way-to-set-a-context-key-and-value/16311

Not directly on point, but seems related (since I'm guessing we want to control which option functions can be emitted).

Also, the original idea for functional options from rpike has them returning private types:

https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html

Using an internal type does have the benefit of preventing callers from defining their own options.

However, better safe than sorry, so added aliasing.

Reply:
Sounds good. Thank you for the additional context!